src/fal/logger.py

```diff
@@ -0,0 +1,119 @@
+from dataclasses import dataclass
+
+from dbt.logger import log_manager
+from dbt.events.functions import fire_event
+from dbt.events.base_types import (
+    TestLevel as _TestLevel,
+    DebugLevel as _DebugLevel,
+    InfoLevel as _InfoLevel,
+    WarnLevel as _WarnLevel,
+    ErrorLevel as _ErrorLevel,
+)
+
+
+class LOGGER:
+    @classmethod
+    def test(cls, msg: str, *args, **kwargs):
+        fire_event(TestMessage(_prepare_msg(msg, *args, **kwargs)))
+
+    @classmethod
+    def trace(cls, msg: str, *args, **kwargs):
+        # Alias to test
+        return cls.test(msg, *args, **kwargs)
+
+    @classmethod
+    def debug(cls, msg: str, *args, **kwargs):
+        fire_event(DebugMessage(_prepare_msg(msg, *args, **kwargs)))
+
+    @classmethod
+    def info(cls, msg: str, *args, **kwargs):
+        fire_event(InfoMessage(_prepare_msg(msg, *args, **kwargs)))
+
+    @classmethod
+    def warn(cls, msg: str, *args, **kwargs):
+        fire_event(WarnMessage(_prepare_msg(msg, *args, **kwargs)))
+
+    @classmethod
+    def warning(cls, msg: str, *args, **kwargs):
+        # Alias to warn
+        return cls.warn(msg, *args, **kwargs)
+
+    @classmethod
+    def error(cls, msg: str, *args, **kwargs):
+        fire_event(ErrorMessage(_prepare_msg(msg, *args, **kwargs)))
+
+
+def _prepare_msg(msg: str, *args, **kwargs):
+    if args:
+        if kwargs:
+            return msg.format(*args, **kwargs)
+        else:
+            return msg.format(*args)
+    else:
+        if kwargs:
+            return msg.format(**kwargs)
+        else:
+            return msg
```

Comment:
ATL: Does this work or is there a scenario where the logic above is needed?
```suggestion
def _prepare_msg(msg: str, *args, **kwargs):
    return msg.format(*args, **kwargs)
 ```

Reply:
There is a scenario. When we log the ipynb files we sometimes get `strings with {braces}` and then the `format` call expects to fill them with something. So we have to test if something WAS passed. I just thought of a small improvement, though.